src/python/WMComponent/WorkflowUpdater/SiteListUpdater.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+"""
+File       : SiteListUpdater
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: provides module to manage update of site lists
+"""
+
+# system modules
+import os
+import json
+
+# WMCore modules
+from WMCore.Services.pycurl_manager import getdata
+from WMCore.WMSpec.WMWorkload import WMWorkloadHelper
+from Utils.CertTools import ckey, cert
+
+class SiteListUpdater(object):
+    def __init__(self, url, logger):
+        """
+        SiteListUpdater provides codebase to update site list info in local agent.
+        Its logic is based on the following:
+        - obtain list of current active workflows from the agent
+        - requests their specs from upstream ReqMgr2 server
+        - update site lists of all workflows
+        - push new specs to the agent database
+
+        :logger: logger to use
+        """
+        self.url = url
+        self.logger = logger
+
+    def getActiveRequests(self, wflowSpecs):
+        """
+        Provide list of active requests in a system (from assigned to running)
+        obtained the agent
+        wflowSpecs = self.listActiveWflows.execute()
+        :wflowSpecs: list of active workflows
+        :return: list of workflows names
+        """
+        requests = []
+        wmaSpecs = {}
+        for wflowSpec in wflowSpecs:
+            name = wflowSpec['name']
+            requests.append(name)
+            wmSpecs[name] = wflowSpec['spec']
+        # TMP: for testing purposes
+        if not requests:
+            requests = [
+                    'apiccine_ReReco_RunBlockWhite_apiccine_StageOutGfalLogs_240930_141359_1693',
+                    'apiccine_ReReco_RunBlockWhite_Nvidia_apiccine_StageOutGfalLogs_240930_141359_9090',
+                    'amaltaro_SC_MultiPU_Aug2024_Val_240730_111429_6786'
+                    ]
+        return requests, wmaSpecs
+
+    def getRequestSpecs(self, requests):
```

Comment:
Even though we are going to need it to fetch the workflow spec itself, I think we will have to implement a call to the `wmstatserver` service and retrieve the site lists with a filter+mask call.

Reply:
Why? if we fetch the whole spec it contains site lists, why do we need to get it again from wmstatserver? I just don't understand the reason.